src/Core/src/Platform/iOS/MauiPageControl.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using CoreGraphics;
+using UIKit;
+
+namespace Microsoft.Maui.Platform.iOS
+{
+	public class MauiPageControl : UIPageControl
+	{
+		const int DefaultIndicatorSize = 6;
+
+		IIndicatorView? _indicatorView;
+		bool _updatingPosition;
+
+		public MauiPageControl()
+		{
+			ValueChanged += MauiPageControlValueChanged;
+			if (NativeVersion.IsAtLeast(14))
+			{
+				AllowsContinuousInteraction = false;
+				BackgroundStyle = UIPageControlBackgroundStyle.Minimal;
+			}
+		}
+		
+		public void SetIndicatorView(IIndicatorView? indicatorView)
+		{
+			if (indicatorView == null)
+			{
+				ValueChanged -= MauiPageControlValueChanged;
+			}
+			_indicatorView = indicatorView;
+
+		}
+
+		public bool IsSquare { get; set; }
+
+		public double IndicatorSize { get; set; }
+
+		protected override void Dispose(bool disposing)
+		{
+			if (disposing)
+				ValueChanged -= MauiPageControlValueChanged;
+			
+			base.Dispose(disposing);
+		}
+
+
+		public override void LayoutSubviews()
+		{
+			base.LayoutSubviews();
+
+			if (Subviews.Length == 0)
+				return;
+
+			UpdateIndicatorSize();
+
+			if (!IsSquare)
+				return;
+
+			UpdateSquareShape();
+		}
+
+		public void UpdateIndicatorSize()
+		{
+			if (IndicatorSize == 0 || IndicatorSize == DefaultIndicatorSize)
+				return;
+
+			float scale = (float)IndicatorSize / DefaultIndicatorSize;
+			var newTransform = CGAffineTransform.MakeScale(scale, scale);
+
+			Transform = newTransform;
+		}
+
+		public void UpdatePosition()
+		{
+			_updatingPosition = true;
+			this.UpdateCurrentPage(GetCurrentPage());
+			_updatingPosition = false;
+
+			int GetCurrentPage()
+			{
+				if (_indicatorView == null)
+					return -1;
+
+				var maxVisible = GetMaximumVisible();
+				var position = _indicatorView.Position;
+				var index = position >= maxVisible ? maxVisible - 1 : position;
+				return index;
+			}
+		}
+
+		public void UpdateIndicatorCount()
+		{
+			this.UpdatePages(GetMaximumVisible());
+			UpdatePosition();
+		}
+
+		void UpdateSquareShape()
+		{
+			if (!NativeVersion.IsAtLeast(14))
+			{
+				UpdateCornerRadius();
+				return;
+			}
+
+			var uiPageControlContentView = Subviews[0];
+			if (uiPageControlContentView.Subviews.Length > 0)
+			{
+				var uiPageControlIndicatorContentView = uiPageControlContentView.Subviews[0];
+
+				foreach (var view in uiPageControlIndicatorContentView.Subviews)
+				{
+					if (view is UIImageView imageview)
+					{
+						imageview.Image = UIImage.GetSystemImage("squareshape.fill");
+						var frame = imageview.Frame;
+						//the square shape is not the same size as the circle so we might need to correct the frame
+						imageview.Frame = new CGRect(frame.X - 6, frame.Y, frame.Width, frame.Height);
```

Comment:
I see a 6, is that the same as the const above?

Reply:
Fixed , removed all this :)